controllers/hostedcontrolplane/hostedcontrolplane_test.go

```diff
@@ -759,10 +760,18 @@ func setupMockServer(handlerFunc http.HandlerFunc) string {
 	return mockServer.URL
 }
 func createMockHandlerFunc(responseBody string, statusCode int) http.HandlerFunc {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(statusCode)
-		w.Write([]byte(responseBody)) // nolint:errcheck
-	})
+	return func(w http.ResponseWriter, r *http.Request) {
+		switch {
+		// Mock GET synthetic monitor response
+		case r.Method == http.MethodGet && r.URL.Path == "/synthetic/monitors/" && r.URL.RawQuery == "tag=cluster-id:mock-cluster-id":
+			w.WriteHeader(http.StatusOK)
+			_, _ = w.Write([]byte(`{"monitors":[{"entityId":"mock-monitor-id"}]}`))
```

Comment:
Is there a reason you can't just pass this byte slice as `responseBody` from your test call?

Reply:
In the way the current code is wrote without the use of interfaces, I have to handle some intermediary http responses for downstream http calls that expect different responses from what this test is expecting. We can consider a interface refactor a future TODO.